src/colvarbias_meta.cpp

```diff
@@ -1782,62 +1779,27 @@ int colvarbias_meta::write_replica_state_file()
   if (cvm::debug()) {
     cvm::log("Writing replica state file for bias \""+name+"\"\n");
   }
-  // write down also the restart for the other replicas
-  cvm::backup_file(replica_state_file.c_str());
-  std::ostream *rep_state_os = cvm::proxy->output_stream(replica_state_file);
+
+  std::string const tmp_state_file(replica_state_file+".tmp");
+  std::ostream *rep_state_os = cvm::proxy->output_stream(tmp_state_file);
   if (rep_state_os == NULL) {
     cvm::error("Error: in opening file \""+
-               replica_state_file+"\" for writing.\n", FILE_ERROR);
+               tmp_state_file+"\" for writing.\n", FILE_ERROR);
     return FILE_ERROR;
   }
 
   rep_state_os->setf(std::ios::scientific, std::ios::floatfield);
 
   if (!write_state(*rep_state_os)) {
     cvm::error("Error: in writing to file \""+
-               replica_state_file+"\".\n", FILE_ERROR);
-    cvm::proxy->close_output_stream(replica_state_file);
+               tmp_state_file+"\".\n", FILE_ERROR);
+    cvm::proxy->close_output_stream(tmp_state_file);
     return FILE_ERROR;
   }
 
-  cvm::proxy->close_output_stream(replica_state_file);
-
-  // rep_state_os.setf(std::ios::scientific, std::ios::floatfield);
-  // rep_state_os << "\n"
-  //              << "metadynamics {\n"
-  //              << "  configuration {\n"
-  //              << "    name " << this->name << "\n"
-  //              << "    step " << cvm::step_absolute() << "\n";
-  // if (this->comm != single_replica) {
-  //   rep_state_os << "    replicaID " << this->replica_id << "\n";
-  // }
-  // rep_state_os << "  }\n\n";
-  // rep_state_os << "  hills_energy\n";
-  // rep_state_os << std::setprecision(cvm::cv_prec)
-  //              << std::setw(cvm::cv_width);
-  // hills_energy->write_restart(rep_state_os);
-  // rep_state_os << "  hills_energy_gradients\n";
-  // rep_state_os << std::setprecision(cvm::cv_prec)
-  //              << std::setw(cvm::cv_width);
-  // hills_energy_gradients->write_restart(rep_state_os);
-
-  // if ( (!use_grids) || keep_hills ) {
-  //   // write all hills currently in memory
-  //   for (std::list<hill>::const_iterator h = this->hills.begin();
-  //        h != this->hills.end();
-  //        h++) {
-  //     rep_state_os << *h;
-  //   }
-  // } else {
-  //   // write just those that are near the grid boundaries
-  //   for (std::list<hill>::const_iterator h = this->hills_off_grid.begin();
-  //        h != this->hills_off_grid.end();
-  //        h++) {
-  //     rep_state_os << *h;
-  //   }
-  // }
-  // rep_state_os << "}\n\n";
-  // rep_state_os.close();
+  cvm::proxy->close_output_stream(tmp_state_file);
+  cvm::backup_file(replica_state_file.c_str());
+  std::rename(tmp_state_file.c_str(), replica_state_file.c_str());
```

Comment:
Check return code of std::rename().

Reply:
I'm looking at `NAMD_backup_file()` to try and be consistent.  Do you recommend to add the extra `remove()` call for Windows and the extra checks for `errno` values?  In that case, it may be useful to just fork `NAMD_backup_file()`.

Reply:
NAMD_backup_file does cover the rare edge cases we've encountered over the years, so forking it would make sense. The big one is looping on EINTR, which NAMD_write also does.

Reply:
Done. Part of this code path still doesn't handle EINTR internally, but most of it is file writes, which do use `NAMD_write()` implicitly.  If this change looks reasonable to you, I'll look later for other places where file I/O could use handling of EINTR.  Thanks.

Reply:
I think this is enough fixing for now.